helpers/metadata.ts

```diff
@@ -70,17 +72,19 @@ function compileSyncTable(syncTable: GenericSyncTable): PackSyncTable {
       listDynamicUrls: compileMetadataFormulaMetadata(listDynamicUrls),
       getter: {
         supportsUpdates: Boolean(executeUpdate),
+        matchers: (matchers || []).map(matcher => matcher.toString()),
```

Comment:
We probably don't have plans to the matchers here on sync tables right? When supporting that like the hackathon project it's probably more intuitive to specify them at the level of the sync table rather than the getter. Our current typing doesn't prevent somebody from providing them on the getter even though they're unused, so I can see why this line is here, but I wonder if we should just omitting the matchers from sync tables until properly supported.

Reply:
Spencer and I were brainstorming about the SDK and I think it's worth having a bit more conversation about where to put matchers on sync tables. 

We thought about how if, e.g., `getSchema` were a property of the formula instead of the table, then the SDK (if not CFL, maybe?) could support dynamic schemas for all formulas. If our proposed `translateUrlIntoDefaultParams` were on the formula, that could allow URL pastes for cards & column formats to also escape the "1 required param" restriction.

We're already dealing with `description` being defined on both the sync table and the formula, and that works out ok, I guess, so maybe we can do something similar for more fields?

Reply:
Yeah for dynamic schemas for regular formulas, it's much more about the CFL and execution challenges than it is about how to model it in the SDK.

One of the reasons it's awkward to put metadata functions onto formulas is that once compiled, they aren't just functions, they are actually formula objects themselves. So if you declare a `getSchema` on your formula, the output is the return value fo `makeMetadataFormula`, which is itself a formula object of the same type as the formula itself. It creates a recursiveness that seems pretty gnarly, though it could possibly still work, and autocomplete metadata formulas are not dissimilar this.

I think the thing to optimize for would be the intuitiveness for the maker. We moved things like the identity name from nested fields to technically-redundant top-level fields because it was important for clarity. Ditto for descriptions--I wouldn't say it's been great to have them at both levels, and I think if our web of types wasn't so nasty, we probably would have typed things so that you could only specify the description at the top level of a sync table.

I think I wouldn't mind moving to a world where `Formula` represents the essential bits of an executable formula, like `parameters`, `resultType`, and `execute` that can be reused as the type for getters and metadata formulas, and `StandaloneFormula` represents an actual building block, and so adds `name`, `description`, `matchers`, etc.